Lib/test/test_smtplib.py

```diff
@@ -549,6 +549,34 @@ def testSendMessageMultipleResentRaises(self):
             smtp.send_message(m)
         smtp.close()
 
+    def testSendMessageAddDateIfMissing(self):
+        m = email.mime.text.MIMEText('A test message')
+        m['From'] = 'foo@bar.com'
+        m['To'] = 'John'
+        m['CC'] = 'Sally, Fred'
+        m['Bcc'] = 'John Root <root@localhost>, "Dinsdale" <warped@silly.walks.com>'
+        smtp = smtplib.SMTP(HOST, self.port, local_hostname='localhost', timeout=3)
+        smtp.send_message(m)
+        # XXX (see comment in testSend)
+        time.sleep(0.01)
+        smtp.quit()
+
+        self.client_evt.set()
+        self.serv_evt.wait()
+        self.output.flush()
+        # The Resent-Bcc headers are deleted before serialization.
+        del m['Bcc']
+        del m['Resent-Bcc']
+        # Add the X-Peer header that DebuggingServer adds
+        m['X-Peer'] = socket.gethostbyname('localhost')
+        mexpect = '%s%s\n%s' % (MSG_BEGIN, m.as_string(), MSG_END)
+        self.assertEqual(self.output.getvalue(), mexpect)
+        debugout = smtpd.DEBUGSTREAM.getvalue()
+        current_moment = email.utils.formatdate()
```

Comment:
Thanks for the review :) I'll work on it tonight, my wife won't be home so I'll have nothing better to do :).

I'm not sure what you mean though on the "mock this up" but I guess you mean to use @mock(email.utils.formatdate) ?

I'll have to read to remember how to do it correctly if it's what you meant.